Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1247,24 +1283,48 @@ void TextureCacheBase::CopyRenderTargetToTexture(u32 dstAddr, EFBCopyFormat dstF
   const u32 bytes_per_row = num_blocks_x * bytes_per_block;
   const u32 covered_range = num_blocks_y * dstStride;
 
-  bool copy_to_ram = !g_ActiveConfig.bSkipEFBCopyToRam;
-  bool copy_to_vram = true;
-
   if (copy_to_ram)
   {
-    EFBCopyParams format(srcFormat, dstFormat, is_depth_copy, isIntensity);
+    EFBCopyParams format(srcFormat, dstFormat, is_depth_copy, isIntensity, y_scale);
     CopyEFB(dst, format, tex_w, bytes_per_row, num_blocks_y, dstStride, srcRect, scaleByHalf);
   }
   else
   {
-    // Hack: Most games don't actually need the correct texture data in RAM
-    //       and we can just keep a copy in VRAM. We zero the memory so we
-    //       can check it hasn't changed before using our copy in VRAM.
-    u8* ptr = dst;
-    for (u32 i = 0; i < num_blocks_y; i++)
+    if (is_xfb_copy)
     {
-      memset(ptr, 0, bytes_per_row);
-      ptr += dstStride;
+      // For XFBs, only Y values between 16 and 236 (and UV values between 16 and 240) are used in
```

Comment:
This following code should likely be moved to its own function, similar to how CopyEFB is separate.

Reply:
Good idea.  In another PR I moved the memory setting for the EFB to another function.  Do you suggest I do that here as well?